### PR TITLE
Add press volume up and down actions

### LINKED ISSUE #2220: Add `volumeUp` and `volumeDown` native capabilities
### Use case

I have test cases where I need to simulate pressing physical volume up and down buttons

### Proposal

Android docs:
- https://developer.android.com/reference/androidx/test/uiautomator/UiDevice#pressKeyCodes(int[])

iOS docs:
- https://developer.apple.com/documentation/xctest/xcuidevice/button/volumeup
- https://developer.apple.com/documentation/xctest/xcuidevice/button/volumedown

### PR DESCRIPTION
**Pull request by pdenert**
Close #2220

### CODE REVIEW

File: packages/patrol/lib/src/native/native_automator2.dart [hunk 1]
```diff
@@ -336,6 +336,16 @@ class NativeAutomator2 {
...
 
+  /// Press volume up
+  Future<void> pressVolumeUp() async {
+    await _wrapRequest('pressVolumeUp', _client.pressVolumeUp);
+  }
+
+  /// Press volume down
+  Future<void> pressVolumeDown() async {
+    await _wrapRequest('pressVolumeDown', _client.pressVolumeDown);
+  }
```
Comment (piotruela) [inline packages/patrol/lib/src/native/native_automator2.dart:347]: Does those methods work on iOS Simulator and Android emulator? If no, please describe the limitation here

File: packages/patrol/lib/src/native/native_automator.dart [hunk 2]
```diff
@@ -452,6 +452,16 @@ class NativeAutomator {
...
+  /// Press volume up
+  Future<void> pressVolumeUp() async {
+    await _wrapRequest('pressVolumeUp', _client.pressVolumeUp);
+  }
+
+  /// Press volume down
+  Future<void> pressVolumeDown() async {
+    await _wrapRequest('pressVolumeDown', _client.pressVolumeDown);
+  }
+
```
Comment (piotruela) [inline packages/patrol/lib/src/native/native_automator.dart:464]: ditto

File: packages/patrol/darwin/Classes/AutomatorServer/Automator/IOSAutomator.swift [hunk 3]
```diff
@@ -346,6 +346,19 @@
       }
     }
 
+    // MARK: Volume settings
+    func pressVolumeUp() throws {
+      runAction("pressing volume up") {
+        self.device.press(XCUIDevice.Button.pressVolumeUp)
+      }
```
Comment (piotruela) [inline packages/patrol/darwin/Classes/AutomatorServer/Automator/IOSAutomator.swift:353]: Acording to the [documentation](https://developer.apple.com/documentation/xctest/xcuidevice/button) this is not available on simulator. I think we should check that before calling just as we do for `runControlCenterAction`

File: packages/patrol/lib/src/native/native_automator2.dart (hunk 1 above)
Comment (pdenert) [reply packages/patrol/lib/src/native/native_automator2.dart]: It works on Android emulator, but doesn't run on iOS Simulator. I've added documentation for that

File: packages/patrol/lib/src/native/native_automator.dart (hunk 2 above)
Comment (pdenert) [reply packages/patrol/lib/src/native/native_automator.dart]: same

**Review by piotruela — APPROVED** [review]
LGTM

**Pull request merged**